pygribjump/src/pygribjump/pygribjump.py

```diff
@@ -140,13 +141,17 @@ def __init__(self, req: dict[str, str], ranges: list[tuple[int, int]], gridHash:
         c_ranges = ffi.new('size_t[]', len(ranges)*2)
         c_ranges_size = len(ranges)*2
         for i, r in enumerate(ranges):
+            if not r[0] < r[1]:
+                raise ValueError(f"Found invalid range {r}: Expected lo < hi for ranges [lo, hi).")
             c_ranges[i*2] = r[0]
             c_ranges[i*2+1] = r[1]
             self.__shape.append(r[1] - r[0])
 
+        self.__c_ranges = c_ranges
```

Comment:
@ChrisspyB I'm new to cffi, do you think this here is an acceptable thing to do or too sketchy and could produce dangling pointers?
In my first version of this PR, I simply stored the original list of ranges as a copy (so just doing `self.__ranges = ranges.copy()` but then was a bit concerned that this introduces unnecessary memory overhead (which *theoretically* might be high for `ExtractionRequest.from_indices` or `ExtractionRequest.from_mask` use-cases).